### ISSUE TITLE
Unable to run on google crowd

### ISSUE THREAD

**Issue by syuntoku14** [crowdAI/marlo-multi-agent-starter-kit#1]
Hi, I am trying to run Minecraft environment on google crowd 16.04 Ubuntu server. It has no GPU.
I tried this solution https://github.com/Microsoft/malmo/issues/291:
```
Install xinit (a tool for manually running an X display server)
Start xinit (e.g., in a separate shell, or detach the process: nohup xinit & disown)
Export the display variable and start the Minecraft client: DISPLAY=:0.0 launchClient.sh
```
but I got the following error:

```
#################################################
         ForgeGradle 2.2-SNAPSHOT-9bc34b4        
  https://github.com/MinecraftForge/ForgeGradle  
#################################################
               Powered by MCP unknown               
             http://modcoderpack.com             
         by: Searge, ProfMobius, Fesh0r,         
         R4wk, ZeuX, IngisKahn, bspkrs           
#################################################
:deobfCompileDummyTask
:deobfProvidedDummyTask
:sourceApiJava UP-TO-DATE
:compileApiJava UP-TO-DATE
:processApiResources UP-TO-DATE
:apiClasses UP-TO-DATE
:copyModToClient
:copyModToServer
:deleteSchemas
:copySchemas
:jaxb UP-TO-DATE
:sourceMainJava UP-TO-DATE
:compileJava
warning: [options] bootstrap class path not set in conjunction with -source 1.6
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
1 warning
:processResources UP-TO-DATE
:classes
:jar
:getVersionJson
:extractNatives SKIPPED
:extractUserdev UP-TO-DATE
:getAssetIndex UP-TO-DATE
:getAssets
Current status: 504/1196   42%
Current status: 976/1196   81%
Current status: 982/1196   82%
Current status: 987/1196   82%
Current status: 1051/1196   87%
:makeStart SKIPPED
:runClient
[06:03:51] [main/INFO] [GradleStart]: Extra: []
[06:03:52] [main/INFO] [GradleStart]: Found and added coremod: com.microsoft.Malmo.OverclockingPlugin
[06:03:52] [main/INFO] [GradleStart]: Running with arguments: [--userProperties, {}, --assetsDir, /home/syuntoku14/.gradle/caches/minecraft/assets, --assetIndex, 1.11, --accessToken{REDACTED}, --version, 1.11.2, --tweakClass, net.minecraftforge.fml.common.launcher.FMLTweaker, --tweakClass, net.minecraftforge.gradle.tweakers.CoremodTweaker]
[06:03:52] [main/INFO] [LaunchWrapper]: Loading tweak class name net.minecraftforge.fml.common.launcher.FMLTweaker
[06:03:52] [main/INFO] [LaunchWrapper]: Using primary tweak class name net.minecraftforge.fml.common.launcher.FMLTweaker
[06:03:52] [main/INFO] [LaunchWrapper]: Loading tweak class name net.minecraftforge.gradle.tweakers.CoremodTweaker
[06:03:52] [main/INFO] [LaunchWrapper]: Calling tweak class net.minecraftforge.fml.common.launcher.FMLTweaker
[06:03:52] [main/INFO] [FML]: Forge Mod Loader version 13.20.0.2228 for Minecraft 1.11.2 loading
[06:03:52] [main/INFO] [FML]: Java is OpenJDK 64-Bit Server VM, version 1.8.0_152-release, running on Linux:amd64:4.15.0-1023-gcp, installed at /home/syuntoku14/anaconda3/envs/marlo/jre
[06:03:52] [main/INFO] [FML]: Managed to load a deobfuscated Minecraft name- we are in a deobfuscated environment. Skipping runtime deobfuscation
[06:03:52] [main/INFO] [FML]: Found a command line coremod : com.microsoft.Malmo.OverclockingPlugin
[06:03:52] [main/WARN] [FML]: The coremod com.microsoft.Malmo.OverclockingPlugin does not have a MCVersion annotation, it may cause issues with this version of Minecraft
[06:03:52] [main/INFO] [LaunchWrapper]: Calling tweak class net.minecraftforge.gradle.tweakers.CoremodTweaker
[06:03:52] [main/INFO] [GradleStart]: Injecting location in coremod net.minecraftforge.fml.relauncher.FMLCorePlugin
[06:03:52] [main/INFO] [GradleStart]: Injecting location in coremod net.minecraftforge.classloading.FMLForgePlugin
[06:03:52] [main/INFO] [GradleStart]: Injecting location in coremod com.microsoft.Malmo.OverclockingPlugin
[06:03:52] [main/INFO] [LaunchWrapper]: Loading tweak class name net.minecraftforge.fml.common.launcher.FMLInjectionAndSortingTweaker
[06:03:52] [main/INFO] [LaunchWrapper]: Loading tweak class name net.minecraftforge.fml.common.launcher.FMLDeobfTweaker
[06:03:52] [main/INFO] [LaunchWrapper]: Loading tweak class name net.minecraftforge.gradle.tweakers.AccessTransformerTweaker
[06:03:52] [main/INFO] [LaunchWrapper]: Calling tweak class net.minecraftforge.fml.common.launcher.FMLInjectionAndSortingTweaker
[06:03:52] [main/INFO] [LaunchWrapper]: Calling tweak class net.minecraftforge.fml.common.launcher.FMLInjectionAndSortingTweaker
[06:03:52] [main/INFO] [LaunchWrapper]: Calling tweak class net.minecraftforge.fml.relauncher.CoreModManager$FMLPluginWrapper
[06:03:52] [main/ERROR] [FML]: The binary patch set is missing. Either you are in a development environment, or things are not going to work!
[06:03:53] [main/ERROR] [FML]: FML appears to be missing any signature data. This is not a good thing
[06:03:53] [main/INFO] [LaunchWrapper]: Calling tweak class net.minecraftforge.fml.relauncher.CoreModManager$FMLPluginWrapper
[06:03:53] [main/INFO] [LaunchWrapper]: Calling tweak class net.minecraftforge.fml.relauncher.CoreModManager$FMLPluginWrapper
[06:03:53] [main/INFO] [LaunchWrapper]: Calling tweak class net.minecraftforge.fml.common.launcher.FMLDeobfTweaker
[06:03:54] [main/INFO] [LaunchWrapper]: Calling tweak class net.minecraftforge.gradle.tweakers.AccessTransformerTweaker
[06:03:54] [main/INFO] [LaunchWrapper]: Loading tweak class name net.minecraftforge.fml.common.launcher.TerminalTweaker
[06:03:54] [main/INFO] [LaunchWrapper]: Calling tweak class net.minecraftforge.fml.common.launcher.TerminalTweaker
[06:03:54] [main/INFO] [LaunchWrapper]: Launching wrapped minecraft {net.minecraft.client.main.Main}
[06:03:54] [main/INFO] [STDOUT]: [com.microsoft.Malmo.OverclockingClassTransformer:transform:58]: MALMO: Attempting to transform MinecraftServer
[06:03:54] [main/INFO] [STDOUT]: [com.microsoft.Malmo.OverclockingClassTransformer:overclockRenderer:187]: MALMO: Found Minecraft, attempting to transform it
[06:03:54] [main/INFO] [STDOUT]: [com.microsoft.Malmo.OverclockingClassTransformer:overclockRenderer:193]: MALMO: Found Minecraft.runGameLoop() method, attempting to transform it
[06:03:54] [main/INFO] [STDOUT]: [com.microsoft.Malmo.OverclockingClassTransformer:overclockRenderer:208]: MALMO: Hooked into call to Minecraft.updateDisplay()
[06:03:55] [main/INFO] [STDOUT]: [com.microsoft.Malmo.OverclockingClassTransformer:transform:42]: net.minecraft.client.entity.EntityPlayerSP
[06:03:55] [main/INFO] [STDOUT]: [com.microsoft.Malmo.OverclockingClassTransformer:transform:42]: net.minecraft.client.entity.AbstractClientPlayer
[06:03:56] [Client thread/INFO] [STDOUT]: [com.microsoft.Malmo.OverclockingClassTransformer:transform:58]: MALMO: Attempting to transform MinecraftServer
[06:03:56] [Client thread/INFO] [STDOUT]: [com.microsoft.Malmo.OverclockingClassTransformer:overclockServer:123]: MALMO: Found MinecraftServer, attempting to transform it
[06:03:56] [Client thread/INFO] [STDOUT]: [com.microsoft.Malmo.OverclockingClassTransformer:overclockServer:129]: MALMO: Found MinecraftServer.run() method, attempting to transform it
[06:03:56] [Client thread/INFO] [STDOUT]: [com.microsoft.Malmo.OverclockingClassTransformer:overclockServer:137]: MALMO: Transforming LDC
[06:03:56] [Client thread/INFO] [STDOUT]: [com.microsoft.Malmo.OverclockingClassTransformer:overclockServer:137]: MALMO: Transforming LDC
[06:03:56] [Client thread/INFO] [STDOUT]: [com.microsoft.Malmo.OverclockingClassTransformer:overclockServer:137]: MALMO: Transforming LDC
[06:03:56] [Client thread/INFO] [STDOUT]: [com.microsoft.Malmo.OverclockingClassTransformer:overclockServer:137]: MALMO: Transforming LDC
[06:03:56] [Client thread/INFO]: Setting user: Player342
[06:04:03] [Client thread/WARN]: Skipping bad option: lastServer:
[06:04:03] [Client thread/INFO]: LWJGL Version: 2.9.4
[06:04:04] [Client thread/INFO]: [STDOUT]: MALMO: Attempting to transform MinecraftServer
[06:04:04] [Client thread/INFO]: [STDOUT]: MALMO: Found GlStateManager, attempting to transform it
[06:04:04] [Client thread/INFO]: [STDOUT]: MALMO: Found GlStateManager.bindTexture() method, attempting to transform it
[06:04:04] [Client thread/INFO]: [STDOUT]: MALMO: Hooked into call to GlStateManager.bindTexture()
[06:04:04] [Client thread/INFO] [STDOUT]: [net.minecraft.init.Bootstrap:printToSYSOUT:600]: ---- Minecraft Crash Report ----
// There are four lights!

Time: 10/30/18 6:04 AM
Description: Initializing game

java.lang.ExceptionInInitializerError
        at net.minecraft.client.Minecraft.setWindowIcon(Minecraft.java:670)
        at net.minecraft.client.Minecraft.init(Minecraft.java:467)
        at net.minecraft.client.Minecraft.run(Minecraft.java:387)
        at net.minecraft.client.main.Main.main(Main.java:118)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
        at net.minecraft.launchwrapper.Launch.main(Launch.java:28)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at net.minecraftforge.gradle.GradleStartCommon.launch(GradleStartCommon.java:97)
        at GradleStart.main(GradleStart.java:26)
Caused by: java.lang.RuntimeException: org.lwjgl.LWJGLException: Could not open X display connection
        at org.lwjgl.opengl.Display.<clinit>(Display.java:141)
        ... 16 more
Caused by: org.lwjgl.LWJGLException: Could not open X display connection
        at org.lwjgl.opengl.LinuxDisplay.openDisplay(Native Method)
        at org.lwjgl.opengl.LinuxDisplay.incDisplay(LinuxDisplay.java:305)
        at org.lwjgl.opengl.LinuxDisplay.internAtom(LinuxDisplay.java:629)
        at org.lwjgl.opengl.LinuxDisplay.init(LinuxDisplay.java:734)
        at org.lwjgl.opengl.Display.<clinit>(Display.java:138)
        ... 16 more


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Client thread
Stacktrace:
        at net.minecraft.client.Minecraft.setWindowIcon(Minecraft.java:670)
        at net.minecraft.client.Minecraft.init(Minecraft.java:467)

-- Initialization --
Details:
Stacktrace:
        at net.minecraft.client.Minecraft.run(Minecraft.java:387)
        at net.minecraft.client.main.Main.main(Main.java:118)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
        at net.minecraft.launchwrapper.Launch.main(Launch.java:28)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at net.minecraftforge.gradle.GradleStartCommon.launch(GradleStartCommon.java:97)
        at GradleStart.main(GradleStart.java:26)

-- System Details --
Details:
        Minecraft Version: 1.11.2
        Operating System: Linux (amd64) version 4.15.0-1023-gcp
        Java Version: 1.8.0_152-release, JetBrains s.r.o
        Java VM Version: OpenJDK 64-Bit Server VM (mixed mode), JetBrains s.r.o
        Memory: 47572392 bytes (45 MB) / 157401088 bytes (150 MB) up to 2075918336 bytes (1979 MB)
        JVM Flags: 1 total; -Xmx2G
        IntCache: cache: 0, tcache: 0, allocated: 0, tallocated: 0
        FML: 
        Loaded coremods (and transformers): 
OverclockingPlugin (MalmoMod-0.36.0.jar)
  com.microsoft.Malmo.OverclockingClassTransformer
        Launched Version: 1.11.2
        LWJGL: 2.9.4
        OpenGL: ~~ERROR~~ RuntimeException: No OpenGL context found in the current thread.
        GL Caps: 
        Using VBOs: No
        Is Modded: Definitely; Client brand changed to 'fml,forge'
        Type: Client (map_client.txt)
        Resource Packs: 
        Current Language: ~~ERROR~~ NullPointerException: null
        Profiler Position: N/A (disabled)
        CPU: <unknown>
[06:04:04] [Client thread/INFO] [STDOUT]: [net.minecraft.init.Bootstrap:printToSYSOUT:600]: #@!@# Game crashed! Crash report saved to: #@!@# /home/syuntoku14/anaconda3/envs/marlo/Minecraft/run/./crash-reports/crash-2018-10-30_06.04.04-client.txt
:runClient FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':runClient'.
> Process 'command '/home/syuntoku14/anaconda3/envs/marlo/bin/java'' finished with non-zero exit value 255

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 36.667 secs
```

How can I solve this issue? 
thanks

**Comment by spMohanty** [comment]
Hi,

Thats because the minecraft client expects an X server or a display
attached.
Some ways in which you can move forward would be to use a VNC server on the
google cloud node you have, or use something like xvfb or xpra for
atacching a fake display to the headless server.

Cheers,
Mohanty

**Comment by syuntoku14** [comment]
Thanks for your reply.

I tried
1. Install OpenGL
2. using Xvfb and launch a fake display
But there was the same error. It seems that the OpenGL doesn't work. 
Do you have any idea?

**Comment by syuntoku14** [comment]
Thank you for your advice.

I used this dockerfile(https://github.com/Microsoft/malmo/blob/master/scripts/docker/build_ubuntu1604_python3.dockerfile) and xvfb, I successfully run 'launchClient.sh' on google cloud.

I don't know the reason but 'sudo xvfb-run -s "-screen 0 1400x900x24" ./launchClient.sh -port 10000' this works fine in a headless server.

To make it easier, I made a docker image to run launchClient.sh in a headless server.
Just run:
```
docker run -it --rm -p [arbitrary port number]:10000 syuntoku/marlo_client
```

It will launch a Minecraft server connected to [arbitrary port number].
